### ISSUE TITLE
Allow moderators to delete messages in chats

### ISSUE THREAD

**Issue by suddenwhipvapor** [bisq-network/bisq2#1897]
I was trying to remove a double question a user had made in two different chats a while ago, and noticed under "more options" that I couldn't delete the other one, but I could report it to myself 😄 
It could be discussed whether it's a good idea to also give mods the privilege to edit messages.

**Comment by HenrikJannsen** [comment]
Technically only the author can remove their message. But we would add a feature for the moderator for flagging messages to be not displayed, thus having the effect like being removed. 
We have too much else atm on our table but I think it would be a useful feature.

**Comment by HenrikJannsen** [comment]
Thinking more about that I think it is not needed. The moderator can ban a user and then all messages of that user are hidden. I think we should use those tools only in severe cases and then its not needed to have more fine-grained features. 
If you agree, feel free to close it, otherwise leave it open.

**Comment by suddenwhipvapor** [comment]
I see the point, and on a flow-mechanics it's not bad as well. A moderator can report a user for whatever reason, just like any other user can. Then if he is lucky he also gets to be the one to judge the report, otherwise it goes to the other mediator who will for sure take into account the judgement by a fellow mediator.
The additional dev load is probably unwarranted.

**Comment by suddenwhipvapor** [comment]
On second thought, while regarding a report to ultimately ban the user, the above makes sense, when instead something else happens, like something that needs to be deleted while the user should not be banned (I think someone inadvertently pasting sensitive information in a public chat, and not being there when a moderator tells him to "remove your previous message") being able to selectively delete single messages whenever is needed is a good thing.
Even in those cases when such necessity should be resolved with a ban anyway, if the report goes to the other mediator it might not be actioned asap and the sensitive information will stay there for everyone to see.
What do you think?

**Comment by HenrikJannsen** [comment]
Yes that case makes sense. But still there is no guarantee that any mod sees the bad content, so damage could stay there for an extended time anyway.

As described initially it would require a new feature for flagging a message by authorized users (mods) and an UI for managing that. Not sure if the effort is worth it. Also we have to be careful with feature bloat. Everything we add has maintenance costs and add complexity. 

There are some other open issues with moderator management (missing notifications,...) which are higher prio IMO and from my part I cannot put it on my table as I want to move on to the next bigger projects like MuSig and mobile. So if any dev wants to work on that they are welcome, but I think it does not represent enough priority for myself.
